src/share/vm/runtime/thread.cpp

```diff
@@ -4704,20 +4734,24 @@ JavaThread *Threads::owning_thread_from_monitor_owner(address owner, bool doLock
 
   {
     MutexLockerEx ml(doLock ? Threads_lock : NULL);
-    ALL_JAVA_THREADS(p) {
-      // first, see if owner is the address of a Java thread
-      if (UseWispMonitor) {
-        if (p->coroutine_list()) {
-          Coroutine* c = p->coroutine_list();
-          do {
-            if ((address) c->wisp_thread() == owner) {
-              return c->wisp_thread();
-            }
-            c = c->next();
-          } while (c != p->coroutine_list());
+    if (UseWispMonitor) {
+      ALL_JAVA_THREADS(p) {
+        Coroutine* c = p->thread_coroutine();
+        if ((address) c->wisp_thread() == owner) {
+          return c->wisp_thread();
+        }
+      }
+      ALL_JAVA_COROUTINES(c) {
```

Comment:
Here I may wonder: around the code we only check about the `Threads_lock`, and the `_coroutine_list_lock` is only used at coroutine insertion and deletion conditions. Do we need it when traversing the global list? Please check about here and other places of `ALL_JAVA_COROUTINES`. If we are traversing the list while some coroutines are inserting in, it may get things broken.